code-journal/src/FormElement.tsx

```diff
@@ -0,0 +1,183 @@
+            <div className="row margin-bottom-1">
+              <div className="column-half">
+                <img
+                  className="input-b-radius form-image"
+                  id="formImage"
+                  src={newImg || (entryEdit?.photoUrl ?? imgSrc)}
+                  alt="image of entry image"
+                  ref={imgRef}
+                />
+              </div>
+              <div className="column-half">
+                <label className="margin-bottom-1 d-block" htmlFor="title">
+                  Title
+                </label>
+                <input
+                  required
+                  className="input-b-color text-padding input-b-radius purple-outline input-height margin-bottom-2 d-block width-100"
+                  type="text"
+                  id="formTitle"
+                  name="formTitle"
+                  value={newTitle || (entryEdit?.title ?? '')}
+                  onChange={(e) => setNewTitle(e.target.value)}
+                />
+                <label className="margin-bottom-1 d-block" htmlFor="photoUrk">
+                  Photo URL
+                </label>
+                <input
+                  required
+                  className="input-b-color text-padding input-b-radius purple-outline input-height margin-bottom-2 d-block width-100"
+                  type="text"
+                  id="formURL"
+                  name="formURL"
+                  value={newImg || (entryEdit?.photoUrl ?? '')}
+                  onChange={(e) => setNewImg(e.target.value)}
+                />
+              </div>
+            </div>
+            <div className="row margin-bottom-1">
+              <div className="column-full">
+                <label className="margin-bottom-1 d-block" htmlFor="formNotes">
+                  Notes
+                </label>
+                <textarea
+                  required
+                  className="input-b-color text-padding input-b-radius purple-outline d-block width-100"
+                  name="formNotes"
+                  id="formNotes"
+                  value={newNote || (entryEdit?.notes ?? '')}
+                  onChange={(e) => setNewNote(e.target.value)}
+                  cols={30}
+                  rows={10}></textarea>
+              </div>
+            </div>
+            <div className="row">
+              <div className="column-full d-flex justify-between">
+                <button
+                  className={`${entryEdit || 'invisible'} delete-entry-button`}
+                  type="button"
+                  id="deleteEntry"
+                  onClick={() => setDeleteConfirmDisplayed(true)}>
+                  Delete Entry
+                </button>
+                <button className="input-b-radius text-padding purple-background white-text">
+                  SAVE
+                </button>
+              </div>
+            </div>
+          </form>
+        </div>
+      </main>
+      {createPortal(
```

Comment:
It would have been nice to create a Popup component like in the react-popup exercise so you'd have something reusable here.